src/sdg_hub/core/flow/base.py

```diff
@@ -591,6 +599,14 @@ def _execute_blocks_on_dataset(
             # Prepare block execution parameters
             block_kwargs = self._prepare_block_kwargs(block, runtime_params)
 
+            block_temp_jsonl_path: Optional[Path] = None
+            dataset_temp_dir: Optional[Path] = None
+            if getattr(block, "_flow_requires_jsonl_tmp", False):
```

Comment:
You're calling `getattr` on block here but haven't checked if `block` has the attribute yet, so it will raise an exception here if it doesn't. Either check `hasattr` first or just directly type the `block` here as the object you expect what you expect and access it directly:


```suggestion
            if hasattr(block, "_flow_requires_jsonl_tmp") and getattr(block, "_flow_requires_jsonl_tmp", False):
```